cmd/kubeadm/app/util/config/cluster_test.go

```diff
@@ -364,6 +364,7 @@ func TestGetAPIEndpointWithBackoff(t *testing.T) {
 			name:     "valid ipv4 endpoint in pod annotation",
 			nodeName: nodeName,
 			staticPod: &testresources.FakeStaticPod{
+				NodeName:  nodeName,
```

Comment:
why are these NodeName changes in the cmd/kubeadm file needed?

Reply:
The fake client being used now respects field selectors, which are being specified by the controller. So the node names need to be specified to be returned by the list/watch.